### ISSUE TITLE
passing object literal as offset to cameraForBounds method will exception

### ISSUE THREAD

**Issue by zhangyiatmicrosoft** [maplibre/maplibre-gl-js#2246]
Repro: pass object literal to cameraForBounds method, like this:
<code>
cameraForBounds([-122.400154,47.512283,-122.266235,47.71672],{"padding": {"left": 600,"right": 300},"offset": {"x": 300,"y": -62}})
</code>

The reason is that at this line:
https://github.com/maplibre/maplibre-gl-js/blob/bc6d6f4c7ed7c0ab71b46f5278edffd6eaeac0a0/src/ui/camera.ts#L605

It did not change the object literal to Point class and line 610, calling "add" method blows up.

Before converting to TS version, this used to work because of this code:
![image](https://user-images.githubusercontent.com/37553549/223029456-8533fd2e-058f-4751-a51c-eae7919d2abe.png)

This block was changed during TS conversion:
https://github.com/maplibre/maplibre-gl-js/pull/209/files#diff-d626aec7548d249604d0505cfaed6100482c49aa039c0d56cbb10ca78a8bad2e

@HarelM Was it intentional? 
More importantly, shall we restore this code to keep backward compatibility? I hope this simple snippet is not bound to MapBox copyright.

**Comment by HarelM** [comment]
The change was suppose to move this logic into the convert method as convert should be responsible for conversion.
I don't fully remember what was the final situation but we started by moving the point class into maplibre and maybe at some point we reverted to use the external library and didn't fix that case.
I would recommend to first add a test to cover this and then maybe create a small utility to convert an object to a point class or send a PR to the point-geometry package.
I don't think this part of the code should be there in that specific method.
I hope this was clear, if not, feel free to ping me.
This is a regression bug nevertheless.

**Comment by drwestco** [comment]
Part of the problem is the type definition for `PointLike`:

```
export type PointLike = Point | [number, number];
```

It doesn't include an arbitrary object with x/y properties.

**Comment by drwestco** [comment]
That this was a breaking change was mentioned in the PR description. #800

**Comment by HarelM** [comment]
Thanks @drwestco for finding the PR. So basically this is work as expected after all...

**Comment by drwestco** [comment]
Well, I wouldn't go quite that far. The exception is expected given the choice made in #800, but from an end-user perspective, having PointLike encompass a plain object with x/y properties would be the reasonable expectation.

**Comment by HarelM** [comment]
True, but this will need an API change, which I'm not against BTW.

**Comment by drwestco** [comment]
Huh. The issue of mapbox/point-geometry supporting bare {x,y} objects was raised in 2016, with a PR submitted in 2018, and ignored since then. Maybe MapLibre shouldn't depend on such an inactive library.

Agreed on this needing an API change. Updating PointLike to support any x/y, and updating all affected APIs would be time better spent than patching just this one instance.

**Comment by HarelM** [comment]
The problem is that we depend on vector-tile package which depends on point, when we tried to remove the dependency we got issue with the generated typescript typings which we didn't manage to solve so we brought this package back and went to create the typings in definitely typed.
Yes, we can fork both and maintain them, but I don't see a really good reason to do so, at least I don't think this issue is a strong enough trigger...

**Comment by zhangyiatmicrosoft** [comment]
Closing the issue for now. Seems like the change was semi-intentional. Even if not, the fix required is quite big and need more discussion. We are going to convince the user to not avoid passing object literal in this case.
